### ISSUE TITLE
EptReader assumes keys in ept.json exist

### ISSUE THREAD

**Issue by kylemann16** [PDAL/PDAL#2546]
Nlohmann Json library has undefined behavior for fetching json keys with unchecked access, ie `m_info["span"]`.

In newer versions of EPT, the `tick` key has been updated to `span`. When using this older version of `ept.json`, PDAL throws an error that doesn't actually indicate what the problem is. The error:

```Error: readers.ept: [json.exception.type_error.302] type must be number, but is number```

This can be fixed by changing lookups like `m_info["span"].get<uint64_t>` in `EptSupport.cpp` into `m_info.at("span").get<uint64_t>`

Pipeline used to replicate error: 
```
[
    {
      "type": "readers.ept",
      "filename": "s3://coordsupport3/autzen",
      "bounds": "([-13700276.746738307,-13699805.433254369],[5473960.653259403,5474431.9667433435],[-106.26672732178122,365.0467566186562])",
      "resolution": 0,
      "threads": 8,
      "tag": "s0"
    },
    {
      "type": "writers.las",
      "filename": "s3://landrush-local/data/54cf1e01-54e9-465b-a55e-580cb1ec7755/clips/af63a8e6-5d84-4ec8-8bd9-40a618bbf6aa/clip.laz",
      "minor_version": 2,
      "dataformat_id": 3,
      "system_id": "Landrush",
      "compression": "laszip",
      "scale_x": 0.01,
      "scale_y": 0.01,
      "scale_z": 0.01,
      "offset_x": -13700277,
      "offset_y": 5473961,
      "offset_z": -106,
      "tag": "s1",
      "inputs": [
        "s0"
      ]
    }
]
```

Similar Error using the sample data from [pdal.io](https://pdal.io/tutorial/iowa-entwine.html) :

```
{
    "pipeline": [
        {
"bounds": "([-10425171.940, -10423171.940], [5164494.710, 5166494.710])",
"filename": "https://s3-us-west-2.amazonaws.com/usgs-lidar-public/IA_FullState",
"type": "readers.ept",
"tag": "readdata"
        },
        {
            "limits": "Classification![7:7]",
            "type": "filters.range",
            "tag": "nonoise"
        },
        {
            "assignment": "Classification[:]=0",
            "tag": "wipeclasses",
            "type": "filters.assign"
        },
        {
            "out_srs": "EPSG:26915",
            "tag": "reprojectUTM",
            "type": "filters.reprojection"
        },
        {
            "tag": "groundify",
            "type": "filters.smrf"
        },
        {
            "limits": "Classification[2:2]",
            "type": "filters.range",
            "tag": "classify"
        },
        {
            "filename": "iowa.laz",
            "inputs": [ "classify" ],
            "tag": "writerslas",
            "type": "writers.las"
        },
        {
            "filename": "iowa.tif",
            "gdalopts": "tiled=yes,     compress=deflate",
            "inputs": [ "writerslas" ],
            "nodata": -9999,
            "output_type": "idw",
            "resolution": 1,
            "type": "writers.gdal",
            "window_size": 6
        }
    ]
}
```

On master branch the error provided is: 
```
pdal: ../vendor/nlohmann/nlohmann/json.hpp:16154: const value_type& nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::operator[](T*) const [with T = const char; ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::const_reference = const nlohmann::basic_json<>&; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::value_type = nlohmann::basic_json<>]: Assertion `m_value.object->find(key) != m_value.object->end()' failed.
```
With the suggested changes, the error is:
```
PDAL: readers.ept: [json.exception.out_of_range.403] key 'points' not found
```

**Comment by abellgithub** [comment]
This is better than current, but still pretty vague for a user.

**Comment by abellgithub** [comment]
Fixed with #2547, though we can do better WRT error feedback.
